pkg/cloud/scope/providerid.go

```diff
@@ -124,3 +125,7 @@ func (p *ProviderID) Validate() bool {
 func (p *ProviderID) IndexKey() string {
 	return p.String()
 }
+
+func ConstructAWSProviderID(instanceID, availabilityZone string) string {
+	return fmt.Sprintf("aws:///%s/%s", availabilityZone, instanceID)
+}
```

Comment:
```suggestion
// ProviderIDPrefix is the prefix of AWS resource IDs to form the Kubernetes Provider ID.
// NOTE: this format matches the 2 slashes format used in cloud-provider and cluster-autoscaler.
const ProviderIDPrefix = "aws://"

// GenerateProviderID generates a valid AWS Node/Machine ProviderID field.
//
// By default, the last id provider is used as identifier (last part)
func GenerateProviderID(ids...string) string {
	return fmt.Sprintf("%s/%s", ProviderIDPrefix, strings.Join(ids, "/")))
}
```

Something like that? Might be useful to add some test cases to match